### ISSUE TITLE
find_package(Thrust CONFIG QUIET) => CUB_DIR-NOTFOUND

### ISSUE THREAD

**Issue by pkestene** [thrust/thrust#1544]
Hello,

I'm using cuda/thrust version 11.4 as provided by Ubuntu (cuda toolkit) 
and my environment is configured like this to detect Thrust:
```shell
export CMAKE_PREFIX_PATH=/usr/local/cuda-11.4/include/cub/cmake:/usr/local/cuda-11.4/include/thrust/cmake"
```

my c++ app build system is detecting Thrust/CUB like this

```cmake
find_package(Thrust QUIET CONFIG)
if (Thrust_FOUND)
    thrust_create_target(Thrust)
endif()
```
But cmake configure actually **succeeds** in detecting Thurst, but **fails** on CUB.

The error reported is:
```txt
CMake Error at /usr/local/cuda-11.4/include/thrust/cmake/thrust-config.cmake:494
CUB_DIR                         *CUB_DIR-NOTFOUND
Thrust_DIR                      */usr/local/cuda-11.4/include/thrust/cmake
```
Two remarks:
- If I manually add `-DCUB_DIR==/usr/local/cuda-11.4/include/cub/cmake` on cmake command line, everything is fine but I shouldn't have to manually set CUB_DIR, it should be detected automatically
- If I look at thust-config.cmake, line 494, actually the fail is due to `find_package(CUB CONFIG ...)` and i think that the hints are missing an additional possibility : "${_THRUST_INCLUDE_DIR}/cub"; if I modify thrust-config.cmake and add this hints, then everything is fine, and I don't need to specify CUB_DIR on cmake command line

I think the hint "${_THRUST_INCLUDE_DIR}/.." is not sufficient, because it expands to `/usr/local/cuda-11.4` and find_package will look for a subdir path that starts with "lib", "share" or "cub" but not "include" (and `cub-config.cmake` is in subdir `include/cub/cmake`).

So my question is:
should this additional hint be added to detect cub, or is there another way to make sure thrust and cub are detected automatically ?
Thank you.
